yaml2rst/rst_writer.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE-SCHEMAS.
+# Copyright (C) 2017 CERN.
+#
+# INSPIRE-SCHEMAS is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE-SCHEMAS is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE-SCHEMAS; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""
+This module defines methods and rules used to parse a ``TreeNode`` to restructured-text strings.
+"""
+
+from __future__ import absolute_import
+from . import rst_util
+import os
+import re
+
+
+PROPERTIES = rst_util.bold('Properties:')
+ANY_OF = 'Can be any element of:'
+ALL_OF = 'Must satisfy the following elements:'
+ONE_OF = 'Must satisfy ONLY one of the following elements:'
+ENUM = rst_util.bold('Elements allowed')
+ITEMS = " elements must be compliant to the following definitions:"
```

Comment:
`'Elements of {this_array_name} must satisfy the following definitions:'` together with `format()` to make this work correctly